PopPUNK/refine.py

```diff
@@ -321,27 +311,23 @@ def growNetwork(sample_names, i_vec, j_vec, idx_vec, s_range, score_idx,
         idx_values = edge_list_df.idx_list.unique()
 
     # Grow a network
-    with tqdm(total=len(idx_values),
-              bar_format="{bar}| {n_fmt}/{total_fmt}",
-              ncols=40,
-              position=thread_idx) as pbar:
+    with tqdm(total = idx_values[-1] + 1,
+              bar_format = "{bar}| {n_fmt}/{total_fmt}",
+              ncols = 40,
+              position = thread_idx) as pbar:
         for idx in idx_values:
             # Create DF
             edge_df = edge_list_df.loc[(edge_list_df['idx_list']==idx),['source','destination']]
```

Comment:
Just checking this slicing is ok efficiency-wise? The index list returned from iter1D is ordered already. But I know slicing is pretty fast (and this is more general, so not sure if now required with these changes)

Reply:
Didn't check - is the best alternative for loop, appending tuples, then generating the network from tuple list or df?

Reply:
Using your meningococcal dataset, dataframe took 69 minutes, loop took 74 minutes - restored and tidied up data frame in aba65b9.

Reply:
Nice!